services/characterService.js

```diff
@@ -0,0 +1,189 @@
+
+class CharacterService {
+  constructor(characterModel, mongoose) {
+    this.CharacterModel = characterModel;
+    this.mongoose = mongoose;
+  }
+
+  async getFullList() {
+    var found = this.CharacterModel.find()
+      .select('-__v')
+      .exec()
+      .then((docs) => {
+        return {
+          count: docs.length,
+          characters: docs.map((doc) => {
+            return {
+              _id: doc._id,
+              name: doc.name,
+              level: doc.level,
+              proficiencyBonus: doc.proficiencyBonus,
+              abilityScores: doc.abilityScores,
+              hp: doc.hp,
+              hpMax: doc.hpMax,
+              description: doc.description,
+              race: doc.raceId,
+              characterClass: doc.characterClassId,
+              request: {
+                type: 'GET',
+                url: 'http://' + process.env.SERVER_ADDRESS + ':' + process.env.PORT + '/characters/' + doc._id
+              }
+            };
+          })
+        };
+      })
+      .catch((err) => {
+        console.log(err);
+        return err;
+      });
+    return found;
+  }
+
+  async getAbbreviatedList() {
+    var found = this.CharacterModel.find()
+      .select('-__v')
+      .exec()
+      .then((docs) => {
+        return {
+          count: docs.length,
+          characters: docs.map((doc) => {
+            return {
+              _id: doc._id,
+              name: doc.name,
+              level: doc.level,
+              request: {
+                type: 'GET',
+                url: 'http://' + process.env.SERVER_ADDRESS + ':' + process.env.PORT + '/characters/' + doc._id
+              }
+            };
+          })
+        };
+      })
+      .catch((err) => {
+        console.log(err);
+        return err;
+      });
+    return found;
+  }
+
+  async createCharacter(body) {
+    const character = new this.CharacterModel({
+      _id: new this.mongoose.Types.ObjectId(),
+      name: body.name,
+      level: body.level,
+      proficiencyBonus: body.proficiencyBonus,
+      hp: body.hp,
+      hpMax: body.hpMax,
+      speed: body.speed,
+      description: body.description,
+      abilityScores: body.abilityScores,
+      race: body.raceId,
+      characterClass: body.characterClassId
+    });
+    const createdChar = character.save()
+      .exec()
```

Comment:
@EmilSchier, I think this is the culprit, you can't call exec() on your model.save() :)

Reply:
Yup, I figured it out